From patchwork Wed Feb 27 02:23:53 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: David Gibson <david@gibson.dropbear.id.au>
X-Patchwork-Id: 10831007
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id BB3471399
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 27 Feb 2019 02:24:03 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A48B42A9E2
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 27 Feb 2019 02:24:03 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 951E22A9F3; Wed, 27 Feb 2019 02:24:03 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham
	version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E66042A9E2
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 27 Feb 2019 02:24:02 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1729774AbfB0CYB (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 26 Feb 2019 21:24:01 -0500
Received: from ozlabs.org ([203.11.71.1]:57947 "EHLO ozlabs.org"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1729610AbfB0CYB (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 26 Feb 2019 21:24:01 -0500
Received: by ozlabs.org (Postfix, from userid 1007)
        id 448KHg2jQQz9s71; Wed, 27 Feb 2019 13:23:59 +1100 (AEDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple;
        d=gibson.dropbear.id.au; s=201602; t=1551234239;
        bh=r0KSHq8hRFPRMBdFC3hM/ggTt0gJiZfRva4AMcNbUOo=;
        h=From:To:Cc:Subject:Date:From;
        b=C/fjlSdEnRyCA4oRiRtZx65lTZEgRwK+t/v0qfyLz9HV2o3IB2CsXX8dmzak3peqU
         2B7qjdOjNkCZMzPmQvoOOW/ikUnZNtw4RsuaWSz6NZvRomxmoovFErfUstX/6EzbWO
         mhUHW6Gn3G1xojjhg9LidXv7sV3alkLspYJzpV4M=
From: David Gibson <david@gibson.dropbear.id.au>
To: lvivier@redhat.com, thuth@redhat.com
Cc: kvm@vger.kernel.org, kvm-ppc@vger.kernel.org,
        David Gibson <david@gibson.dropbear.id.au>
Subject: [PATCH] pwoerpc: Enable HTM capability for h_cede_tm testcase
Date: Wed, 27 Feb 2019 13:23:53 +1100
Message-Id: <20190227022354.12726-1-david@gibson.dropbear.id.au>
X-Mailer: git-send-email 2.20.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The h_cede_tm case is specifically about an edge case with the H_CEDE
hypercall while using hardware transactional memory (HTM).  Current qemu
versions don't enable HTM instructions in guests by default, so we should
explicitly enable them in order to execute this test.

Signed-off-by: David Gibson <david@gibson.dropbear.id.au>
Reviewed-by: Thomas Huth <thuth@redhat.com>
---
 powerpc/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/powerpc/unittests.cfg b/powerpc/unittests.cfg
index 4eda258..af535b7 100644
--- a/powerpc/unittests.cfg
+++ b/powerpc/unittests.cfg
@@ -63,7 +63,7 @@ file = emulator.elf
 [h_cede_tm]
 file = tm.elf
 smp = 2,threads=2
-extra_params = -append "h_cede_tm"
+extra_params = -machine cap-htm=on -append "h_cede_tm"
 groups = nodefault,h_cede_tm
 
 [sprs]
